### DISCUSSION TITLE
Order a bot to construct an item

### DISCUSSION THREAD

**Discussion by Xayrlen** [FakeFishGames/Barotrauma#12892] [Ideas and suggestions]
I'm a singleplayer enjoyer and I often wish that I could make my bot-crew do something else aside from fixing stuff, manning turrets and walking around the sub.

I could imagine adding a specific order to the existing menu, where you choose a crewmate and order them to construct an item (a standard constructor menu will open up and bot will begin, if necessary materials are present), while I am driving. It would save up a lot of time and make less distractions while on mission.

**Comment by bookslayer10** [comment]
If making a whole interface to tell the bots what to craft is too much, I think a good alternative would be an order to get the bots to fabricate items on your behalf. You would order a bot down to the fabricator to await crafting instructions. Then, anyone could go down there and use the fabricator interface to start crafting something, but the bot that was ordered to man the fabricator would be the one that is actually doing the crafting. For example, it would use the crafting skills and talents of the bot when determining item quality, and any skill gain would go to the bot manning the station, not the person who pressed the button.

It's a decent alternative that can still save quite a lot of time, while being easier to implement than the functionality for telling the bots to craft things remotely. Ideally, we would get a "craft this item" command, but a "help me craft items" command would also be nice.

**Comment by BlackMoons** [comment]
This honestly has a few major uses.

In singleplayer, I don't have time to be manning the assembler, churning out railgun shells (Maybe NPC could attempt to auto-store item via 'cleanup items' routine after its made), and don't really want to sit at a station doing some of those more tedius tasks. 

In multiplayer, you can't switch to bots, so it would be essential to be able to use a bots crafting skill or crafting talents (like increased quality, or the 'don't use 1 input ingredient' talent, etc)

**Comment by gerakis** [comment]
Just adding a queue to the fabricator would do the same thing much simpler. If I could make 32 railgun shells the same way I can make 32 oxygen tanks this wouldn't even be an issue.
